control/iosys.py

```diff
@@ -918,30 +925,46 @@ def __init__(self, syslist, connections=[], inplist=[], outlist=[],
             system_count += 1
 
         # Check for duplicate systems or duplicate names
-        sysobj_list = []
-        sysname_list = []
+        # Duplicates are renamed sysname_1, sysname_2, etc.
+        sysobj_name_dct = {}
+        sysname_count_dct = {}
         for sys in syslist:
-            if sys in sysobj_list:
-                warn("Duplicate object found in system list: %s" % str(sys))
-            elif sys.name is not None and sys.name in sysname_list:
-                warn("Duplicate name found in system list: %s" % sys.name)
-            sysobj_list.append(sys)
-            sysname_list.append(sys.name)
+            if sys in sysobj_name_dct:
+                raise ValueError("Duplicate object found in system list: %s" % str(sys))
+            elif sys.name is not None and sys.name in sysname_count_dct:
+                count = sysname_count_dct[sys.name]
+                sysname_count_dct[sys.name] += 1
+                sysname = sys.name + "_" + str(count)
+                sysobj_name_dct[sys] = sysname
+                warn("Duplicate name found in system list. Renamed to {}".format(sysname))
+            else:
+                sysname_count_dct[sys.name] = 1
+                sysobj_name_dct[sys] = sys.name
+                
+        # This won't work when there are duplicate sysnames
+        # Should we throw an error rather than just warning above?
+        # Users can use sys.copy() to get a unique name for the dup system.
```

Comment:
Now obsolete, right? Maybe document the sys.copy part it in the function docstring?

Reply:
Done. sys.copy() is no longer necessary, so no need to add to docstring.